debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -277,8 +283,22 @@ public void commit() throws InterruptedException {
 
         if (locked) {
             try {
-                if (coordinator != null && lastOffset != null) {
-                    coordinator.commitOffset(lastOffset);
+                if (coordinator != null) {
+                    Iterator<Map.Entry<Map<String, ?>, Map<String, ?>>> iterator = lastOffsets.entrySet().iterator();
+                    while (iterator.hasNext()) {
```

Comment:
Could this be simplified to `for (Map.Entry<..., ...> entry : lastOffsets.entrySet())`?